### PR TITLE
Create ABBYY file watcher

### PR DESCRIPTION
**Pull request by thatbudakguy**
- Create a workflow updater service for text extraction
- Create a FileWatcher that emits events based on ABBYY jobs

#1231 will create a background service to run the `FileWatcher` using systemd.

### CODE REVIEW

File: lib/dor/text_extraction/abbyy/file_watcher.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        # These methods control the underlying listener; see:
+        # https://github.com/guard/listen?tab=readme-ov-file#pause--start--stop
+        # NOTE: You need to `sleep` or otherwise avoid letting the process
+        # exit after starting in order for the listener's thread to keep running
+        delegate :start, :pause, :stop, to: :listener
+
+        # Directories where ABBYY places OCR result information
+        RESULT_XML_DIR = 'RESULTXML'
+        EXCEPTIONS_DIR = 'EXCEPTIONS'
```
Comment (dnoneill) [inline lib/dor/text_extraction/abbyy/file_watcher.rb:18]: These should be set in settings.

File: lib/dor/text_extraction/abbyy/file_watcher.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        def initialize(root_path:, workflow_updater: nil, listener_options: {})
+          @root_path = root_path
+          @result_xml_path = File.join(root_path, RESULT_XML_DIR)
+          @exceptions_path = File.join(root_path, EXCEPTIONS_DIR)
```
Comment (dnoneill) [inline lib/dor/text_extraction/abbyy/file_watcher.rb:23]: See above. Entire path should be in settings.

File: lib/dor/text_extraction/abbyy/file_watcher.rb (hunk 1 above)
Comment (thatbudakguy) [reply lib/dor/text_extraction/abbyy/file_watcher.rb:18]: Didn't realize we had that! Thanks

Comment (dnoneill) [reply lib/dor/text_extraction/abbyy/file_watcher.rb:18]: Actually it's in windows. Might not be good.

Comment (thatbudakguy) [reply lib/dor/text_extraction/abbyy/file_watcher.rb]: That might be OK if we set it to the windows version in production/on the VMs though, right? But we can leave it non-windows in local dev?

Comment (dnoneill) [reply lib/dor/text_extraction/abbyy/file_watcher.rb]: Well it's only windows on the Abbyy vm, but the common-accessing vm mount is unix and at a different place.

Comment (thatbudakguy) [reply lib/dor/text_extraction/abbyy/file_watcher.rb]: Ahh, and we use this setting for two different purposes? Maybe we need two settings then, or something...

File: lib/dor/text_extraction/abbyy/file_watcher.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        end
+
+        # Notify SDR that the OCR workflow step completed successfully
+        def process_success(results)
+          @workflow_updater.mark_ocr_completed(results.druid)
+        end
+
+        # Notify SDR that the OCR workflow step failed
+        def process_failure(results)
+          @workflow_updater.mark_ocr_errored(results.druid)
```
[the remark this answers is not in the document]
Comment (thatbudakguy) [reply lib/dor/text_extraction/abbyy/file_watcher.rb:56]: That's a good idea; I'll look for an example of how to do that via the workflow service.

[the remark this answers is not in the document]
Comment (thatbudakguy) [reply lib/dor/text_extraction/abbyy/file_watcher.rb]: Added some stuff to report using the error messages that `Results` knows about.

File: lib/dor/text_extraction/abbyy/file_watcher.rb (hunk 1 above)
Comment (thatbudakguy) [reply lib/dor/text_extraction/abbyy/file_watcher.rb]: I split out the two path styles into separate values, and created https://github.com/sul-dlss/shared_configs/pull/2183.

**Review by peetucket — APPROVED** [review]

**Pull request merged**